app/lib/pdf_filler/md502_r_pdf.rb

```diff
@@ -0,0 +1,24 @@
+module PdfFiller
+  class Md502RPdf
+    include PdfHelper
+
+    def source_pdf_name
+      "md502R-TY2023"
+    end
+
+    def initialize(submission)
+      @submission = submission
+
+      # Most PDF fields are grabbed right off the XML
+      builder = StateFile::StateInformationService.submission_builder_class(:md)
+      @xml_document = builder.new(submission).document
+      @calculator = submission.data_source.tax_calculator
```

Comment:
we shouldn't call the calculator here, the only place the calculator should be called is in the submission builder. Instead we should grab pdf elements from the `@xml_document` directly or intake/df_data when its not present in the XML.